src/buybutton.js

```diff
@@ -15,13 +15,13 @@ ShopifyBuy.UI = window.ShopifyBuy.UI || {
   domains: {},
 
   init(client, integrations = {}, styleOverrides) {
-    const domain = client.shop.fetchInfo().then((res) => {
```

Comment:
@harisaurus turns out this was a long standing bug breaking any multi-shop set ups.